pkg/resources/database/resource_db_pg.go

```diff
@@ -0,0 +1,310 @@
+func (p *PGDatabaseResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+
+		MarkdownDescription: "❗ Manage service database for a PostgreSQL Exoscale [Database Services (DBaaS)](https://community.exoscale.com/documentation/dbaas/).",
+		Attributes: map[string]schema.Attribute{
+			// Computed attributes
+			"id": schema.StringAttribute{
+				MarkdownDescription: "The ID of this resource, computed as service/database_name",
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			// Attributes referencing the service
+			"service": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "❗ The name of the database service.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"zone": schema.StringAttribute{
+				MarkdownDescription: "❗ The Exoscale [Zone](https://www.exoscale.com/datacenters/) name.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf(config.Zones...),
+				},
+			},
+			// Variables
+			"database_name": schema.StringAttribute{
+				Required:            true,
+				MarkdownDescription: "❗ The name of the database for this service.",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"lc_collate": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "Default string sort order (LC_COLLATE) for PostgreSQL database",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"lc_ctype": schema.StringAttribute{
+				Optional:            true,
+				MarkdownDescription: "Default character classification (LC_CTYPE) for PostgreSQL database",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+		},
+		Blocks: map[string]schema.Block{
+			"timeouts": timeouts.BlockAll(ctx),
+		},
+	}
+}
+
+// Update implements resource.Resource.
+func (p *PGDatabaseResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var stateData, planData PGDatabaseResourceModel
+	UpdateResource(ctx, req, resp, &stateData, &planData, p.client)
+}
+
+// ReadResource reads resource from remote and populate the model accordingly
+func (data PGDatabaseResourceModel) ReadResource(ctx context.Context, client *v3.Client, diagnostics *diag.Diagnostics) {
+	svc, err := client.GetDBAASServicePG(ctx, data.Service.ValueString())
+	if err != nil {
+		diagnostics.AddError("Client Error", fmt.Sprintf("Unable to read service pg, got error: %s", err))
+		return
+	}
+
+	for _, db := range svc.Databases {
+		if string(db) == data.DatabaseName.ValueString() {
+			return
```

Comment:
Shouldn't you set `lc_collate` and `lc_ctype` here?

Reply:
The information is not retrievable by the API so we don't know which values to update it to. I added a comment to make that clear.